superset/db_engine_specs/presto.py

```diff
@@ -25,23 +25,27 @@
 from typing import Any, Dict, List, Optional, Set, Tuple
 from urllib import parse
 
+import simplejson as json
 from sqlalchemy import Column, literal_column
 from sqlalchemy.engine.base import Engine
 from sqlalchemy.engine.reflection import Inspector
 from sqlalchemy.engine.result import RowProxy
 from sqlalchemy.sql.expression import ColumnClause, Select
 
-from superset import is_feature_enabled
+from superset import app, is_feature_enabled, security_manager
 from superset.db_engine_specs.base import BaseEngineSpec
 from superset.exceptions import SupersetTemplateException
 from superset.models.sql_types.presto_sql_types import type_map as presto_type_map
+from superset.sql_parse import ParsedQuery
 from superset.utils import core as utils
 
 QueryStatus = utils.QueryStatus
+config = app.config
 
 
 class PrestoEngineSpec(BaseEngineSpec):
     engine = "presto"
+    allows_cost_estimate = True
```

Comment:
What version of presto was this added in? Do we need to check the version number to make sure this is available?

Reply:
The PR has been merged, but It hasn't been released yet. We patched our Presto cluster with the commit and have it live.

Also, I'm not sure it's trivial to have features depending on the DB version, I remember asking @john-bodley about this for Druid.

Reply:
I implemented this already in another part of the presto engine spec, relying on the user specifying the version in the extra data field of the db model, but falling back to assuming the most recent version. check out: https://github.com/apache/incubator-superset/blob/master/superset/db_engine_specs/presto.py#L918

Reply:
Ah, nice. Let me take a look.

Reply:
Cool, I'll remove the DB migration and make it configurable via extras, as well as using the version to see if the DB supports cost estimation.